gridcomps/History/MAPL_HistoryTrajectoryMod_smod.F90

```diff
@@ -1078,41 +1200,41 @@
            rT1=real(i1, kind=ESMF_KIND_R8)
            rT2=real(i2, kind=ESMF_KIND_R8)
            jlo = 1
-           jhi= size(this%obstime)
+           !!
+           !! I choose UB = N+1 not N, because my sub. bisect find n: Y(n)<x<=Y(n+1)
+           jhi= size(this%obstime) + 1
            if (jhi==0) then
               x_subset(1:2)=0
               _RETURN(_SUCCESS)
            endif
 
            lb=int(jlo, ESMF_KIND_I8)
            ub=int(jhi, ESMF_KIND_I8)
-           call bisect( this%obstime, rT1, jt1, n_LB=lb, n_UB=ub, rc=rc)
-           call bisect( this%obstime, rT2, jt2, n_LB=lb, n_UB=ub, rc=rc)
-           x_subset(1) = jt1
-
-           if (jt1<lb) then
-              if (jt2<lb) then
-                 x_subset(1) = 0
-                 x_subset(2) = 0
-              elseif (jt2>=ub) then
-                 x_subset(1) = lb
-                 x_subset(2) = ub
-              else
-                 x_subset(1) = lb
-                 x_subset(2) = jt2
-              endif
-           elseif (jt1>=ub) then
-              x_subset(1) = 0
-              x_subset(2) = 0
-           else
-              x_subset(1) = jt1
-              if (jt2>=ub) then
-                 x_subset(2) = ub
-              else
-                 x_subset(2) = jt2
-              endif
-           endif
-
+           call bisect( this%obstime, rT1, index1, n_LB=lb, n_UB=ub, rc=rc)
+           call bisect( this%obstime, rT2, index2, n_LB=lb, n_UB=ub, rc=rc)
+
+           ! (x1, x2]  design in bisect
+           !  simple version
+
+           x_subset(1) = index1+1
+           x_subset(2) = index2
+
+!           if (index1<lb) then
+!              if (index2<lb) then
+!                 x_subset(2) = index2
+!              elseif (index2>=ub) then
+!                 x_subset(2) = ub
+!              else
+!                 x_subset(2) = index2
+!              endif
+!           elseif (index1>=ub) then
+!              x_subset(1) = 0
+!              x_subset(2) = 0
+!           else
+!              x_subset(2) = index2
+!           endif
+!
```

Comment:
Dead code?

Reply:
Deleted.   The effect will show up in another PR.